src/cargo/core/compiler/fingerprint.rs

```diff
@@ -1547,6 +1550,9 @@ impl DepInfoPathType {
 /// The `rustc_cwd` argument is the absolute path to the cwd of the compiler
 /// when it was invoked.
 ///
+/// If the `allow_package` argument is false, then package-relative paths are
+/// skipped and ignored.
```

Comment:
Could this comment be expanded to include rationale for why we'd set it to `false`?

It looked a bit odd for a second filtering here vs filtering when we parse the file, but I don't see any reason why we'd do it the other way around